tutorials/realm/Barrier.md

```diff
@@ -0,0 +1,178 @@
+---
+layout: page
+permalink: /tutorial/barrier.html
+title: Realm Barrier
+---
+
+## Introduction
+
+A `Barrier` is a special kind of `Event` (notice that it inherits from the base Event class) that 
+requires multiple arrivals in order for it to trigger. Additionally, unlike normal events, 
+which only have a single generation, a barrier has multiple generations, named phases. 
+Therefore, a barrier can be advanced to the next generation and performed arrivals 
+even before the previous generation has been triggered. However, a barrier has to be explicitly destroyed. 
+In this example, there are two types of tasks, reader and writer. 
+During each iteration, every writer task generates an integer value. Subsequently, 
+the collective sum of all these integers is calculated, and the resulting sum is then accessed by the reader tasks. 
+Thus, the writer and reader tasks take turns to execute.
+We illustrate how to use `Barrier` to go back and forth between 
+reader and writer tasks of different iterations.
+
+Here is a list of covered topics:
+
+* [Creating Barriers](#creating-barriers)
+* [Creating Reductions](#creating-reductions)
+* [Synchronizing Tasks with Barriers](#synchronizing-tasks-with-barriers)
+* [Performance Considerations](#performance-considerations)
+* [Limitations](#limitations)
+
+## Creating Barriers
+
+As introduced in the first section, each writer task performs a reduction operation on its local integer value. 
+The reader tasks can not read the reduction value until all writer tasks have completed writing. 
+This kind of synchronization can be implemented by `Barrier`. 
+In this example, we need to create two barriers, a `writer_barrier` and `reader_barrier`,
+to synchronize writer and reader tasks, respectively. 
+To create a barrier, we need to pass the number of concurrent barrier tasks into the `create_barrier` function.
+The following code demonstrates creating a `reader_barrier`, where we need to pass the number of participants:
+```c++
```

Comment:
In general, please add a blank line before all code fences. Not all Markdown parsers handle this uniformly and it's best to make our Markdown as compatible as possible so we don't risk having issue in the future.

Reply:
fixed for all realm tutorials.